sale_backorder/models/sale.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    last_date_delivered = fields.Datetime(
+        string="Last Date Delivered", compute="_compute_last_date_delivered", store=True
+    )
+    last_bill_date = fields.Datetime(
+        string="Last Bill Date", compute="_compute_last_bill_date", store=True
+    )
+    uigd_value = fields.Monetary(
+        string="UIGD Value", compute="_compute_uigd_value", store=True
+    )
+    bo_value = fields.Monetary(
+        string="Backorder Value", compute="_compute_bo_value", store=True
+    )
+
+    @api.depends("order_line.uigd_qty", "order_line.price_unit")
+    def _compute_uigd_value(self):
+        for order in self:
+            total = 0
+            for line in order.order_line:
+                total += line.uigd_value
+            order.uigd_value = total
+
+    @api.depends("order_line.bo_qty", "order_line.price_unit")
+    def _compute_bo_value(self):
+        for order in self:
+            total = 0
+            for line in order.order_line:
+                total += line.bo_value
+            order.bo_value = total
+
+    @api.depends(
+        "order_line.last_date_delivered",
+        "order_line.qty_delivered",
+        "order_line.qty_invoiced",
+    )
+    def _compute_last_date_delivered(self):
+        for order in self:
+            max_date = False
+            for line in order.order_line:
+                if line.last_date_delivered:
+                    max_date = line.last_date_delivered
+            order.last_date_delivered = max_date
+
+    @api.depends(
+        "order_line.last_bill_date",
+        "order_line.qty_delivered",
+        "order_line.qty_invoiced",
+    )
+    def _compute_last_bill_date(self):
+        for order in self:
+            max_date = False
+            for line in order.order_line:
+                if line.last_bill_date:
+                    max_date = line.last_bill_date
+            order.last_bill_date = max_date
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    last_date_delivered = fields.Datetime(
+        string="Last Date Delivered", compute="_compute_last_date_delivered", store=True
+    )
+    last_bill_date = fields.Datetime(
+        string="Last Bill Date", compute="_compute_last_bill_date", store=True
+    )
+    uigd_qty = fields.Float(string="UIGD Qty", compute="_compute_uigd_qty", store=True)
```

Comment:
IMHO explicit field names is better

Reply:
@rousseldenis i.e field name is Like un_Invoiced_goods_delivered ??

Reply:
Put yourself at the place of user/developer that came after.

uninvoiced_goods_delivered_qty. I suppose string is to help display in tree view. But you should add a help on that field too.